t/ids_from_pages.rb

```diff
@@ -27,4 +27,40 @@
       end
     end
   end
+
+  describe 'Redirect links' do
+    describe 'single redirect link' do
+      let(:titles) { ['The Deeper Meaning of Liff'] }
+
+      it 'pairs link with wikidata ID' do
+        subject.must_equal('The Deeper Meaning of Liff' => 'Q875382')
+      end
+    end
+
+    describe 'single redirect and single direct link' do
+      let(:titles) { ['The Deeper Meaning of Liff', 'The Meaning of Liff'] }
+
+      it 'pairs both redirecting link and direct link with wikidata ID' do
+        subject.must_equal('The Deeper Meaning of Liff' => 'Q875382', 'The Meaning of Liff' => 'Q875382')
+      end
+    end
+
+    describe 'multiple redirect links' do
+      let(:titles) { ['The Deeper Meaning of Liff', 'Marvin the Paranoid Android'] }
+
+      it 'returns wikidata ids paired with redirecting links and direct links' do
+        subject.must_equal('The Deeper Meaning of Liff'  => 'Q875382',
+                           'Marvin the Paranoid Android' => 'Q264685')
+      end
+    end
+
+    describe 'multiple links redirecting to the same article' do
+      let(:titles) { ['Wikkit Gate', 'Infinite Improbability Drive'] }
+
+      it 'returns wikidata ids paired with redirecting links and direct links' do
+        subject.must_equal('Wikkit Gate'                  => 'Q259299',
+                           'Infinite Improbability Drive' => 'Q259299')
```

Comment:
The implementation here seems a little too lossy — this is throwing away that these are redirecting to "Technology in The Hitchhiker's Guide to the Galaxy"